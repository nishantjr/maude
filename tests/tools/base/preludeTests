#!/bin/sh

MAUDE_LIB=$srcdir/../../../src/Main
export MAUDE_LIB

../../../src/Main/maude \
  < preludeTests.maude -no-banner -no-advise \
  > preludeTests.out 2>&1

diff preludeTests.expected preludeTests.out > /dev/null 2>&1
