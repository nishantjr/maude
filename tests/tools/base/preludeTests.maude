load ../../../contrib/tools/base/prelude.maude

--- Module BOUND
--- ------------

select BOUND .

reduce not (unbounded < 3) .
reduce 3 < unbounded .
reduce (unbounded < unbounded =/= true) and (unbounded < unbounded =/= false) .

reduce continue(3, 5)         == 4, 5 .
reduce continue(3, unbounded) == 4, unbounded .
reduce      continue(3, 4)                 :: Depth .
reduce      continue(3, unbounded)         :: Depth .
reduce not (continue(unbounded, unbounded) :: Depth) .
reduce not (continue(5, 3)                 :: Depth) .
