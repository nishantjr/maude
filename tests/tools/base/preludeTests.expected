==========================================
reduce in BOUND : not unbounded < 3 .
rewrites: 3 in 0ms cpu (0ms real) (~ rewrites/second)
result Bool: true
==========================================
reduce in BOUND : 3 < unbounded .
rewrites: 1 in 0ms cpu (0ms real) (~ rewrites/second)
result Bool: true
==========================================
reduce in BOUND : unbounded < unbounded =/= true and unbounded < unbounded =/=
    false .
rewrites: 3 in 0ms cpu (0ms real) (~ rewrites/second)
result Bool: true
==========================================
reduce in BOUND : continue(3,5) == 4,5 .
rewrites: 3 in 0ms cpu (0ms real) (~ rewrites/second)
result Bool: true
==========================================
reduce in BOUND : continue(3,unbounded) == 4,unbounded .
rewrites: 3 in 0ms cpu (0ms real) (~ rewrites/second)
result Bool: true
==========================================
reduce in BOUND : continue(3,4) :: Depth .
rewrites: 3 in 0ms cpu (0ms real) (~ rewrites/second)
result Bool: true
==========================================
reduce in BOUND : continue(3,unbounded) :: Depth .
rewrites: 3 in 0ms cpu (0ms real) (~ rewrites/second)
result Bool: true
==========================================
reduce in BOUND : not continue(unbounded,unbounded) :: Depth .
rewrites: 3 in 0ms cpu (0ms real) (~ rewrites/second)
result Bool: true
==========================================
reduce in BOUND : not continue(5,3) :: Depth .
rewrites: 4 in 0ms cpu (0ms real) (~ rewrites/second)
result Bool: true
Bye.
